### ISSUE TITLE
"Must use import to load ES Module" when trying to use ES module in Jest

### ISSUE THREAD

**Issue by ColdSpirit0** [alangpierce/sucrase#741]
Hi! When I trying to test code that imports ES modules, it fails with error:

```
> node --experimental-vm-modules node_modules/jest/bin/jest.js

(node:5112) ExperimentalWarning: VM Modules is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
 FAIL  tests/index.test.ts
  ● Test suite failed to run

    Must use import to load ES Module: D:\Work\js\test_modules\node_modules\is-path-inside\index.js

    > 1 | import isPathInside from "is-path-inside";
        | ^
      2 |
      3 | export function fun1() {
      4 |     return isPathInside("/aaa/bbb", "/aaa")

      at Runtime.requireModule (node_modules/jest-runtime/build/index.js:1011:21)
      at Object.<anonymous> (src/index.ts:1:1)
      at Object.<anonymous> (tests/index.test.ts:1:1)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        0.76 s
Ran all test suites.
```

Any idea how to fix that?
Example project: https://github.com/ColdSpirit0/jest-sucrase-problem

**Comment by alangpierce** [comment]
Hi @ColdSpirit0 , thanks for reporting and for setting up the example project! I took a look and it looks like the Jest integration needs to be updated to read the Jest options and preserve `import` statements if Jest is running in ESM mode. Should be a relatively straightforward fix I think, and I'll leave this issue open for tracking.

**Comment by alangpierce** [comment]
I just released a new version of the Jest plugin as 3.0.0 with a fix for this. When Jest is configured to treat a file as an ES module, it passes that information into the plugin, which invokes Sucrase with the right configuration. Note that you may need to specify `extensionsToTreatAsEsm: ['.ts'],` in the Jest option to get Jest to treat TS files as ESM.
